readme.md

```diff
@@ -198,17 +209,36 @@ const ctx = {
   country: 'canada',
   age: 21,
   options: [1,2,3],
+  optionIndex: [1],
   address: {
     city: 'Toronto',
     country: 'Canada',
-  }
+  },
+  address2: {
+    city: 'Vancouver',
+    country: 'Canada',
+  },
+  selectedAddress: '2',
+  child: 'country'
 }
 
 // Evaluate an expression in the given data context
 engine.evaluate(['>', '$age', 20], ctx); // true
 
 // Evaluate an expression in the given data context
 engine.evaluate(['==', '$address.city', 'Toronto'], ctx); // true
+
+// Evaluate an expression using a reference that includes the array notation
+engine.evaluate(['==', '$options[1]', 2], ctx) // true
+
+// Evaluate an expression using a complex array reference
+engine.evaluate(['==', '$options[{optionIndex}]', 2], ctx) // true
```

Comment:
since optionIndex is `[1]`, wouldn't this evaluate to `$options[[1]]`, and is that the desired?

Reply:
my bad, optionIndex should just be `1`.

Changing this as per David's comment

Reply:
13b35e60462b79c6125c239e32eb7acd8406cdba